Description: Transition from QT4 to QT5
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2015-05-03
Forwarded: no

Index: zygrib-7.0.0/Makefile
===================================================================
--- zygrib-7.0.0.orig/Makefile
+++ zygrib-7.0.0/Makefile
@@ -6,14 +6,12 @@ QWTDIR=src/qwt-6.1.3
 
 all: zyGrib
 
-SRC= src/*.h src/*.cpp src/*/*.h src/*/*.cpp \
-      src/qwt*/src/*.cpp src/qwt*/src/*.h
+SRC= src/*.h src/*.cpp src/*/*.h src/*/*.cpp 
 
 mac: $(SRC)
 	rm -f  ./zyGrib
 	rm -fr ./zyGrib.app
 	rm -f  src/Makefile
-	cd src/qwt-6.0.1/src; $(MACQTBIN)/qmake; make -j6
 	cd src; $(MACQTBIN)/qmake; make -j6
 
 macclean:
@@ -21,7 +19,6 @@ macclean:
 	rm -fr zyGrib.app
 	rm -f  src/zyGrib
 	rm -f  src/release/zyGrib.exe
-	cd src/qwt-6.0.1/src; $(MACQTBIN)/qmake; make clean
 	cd src;	$(MACQTBIN)/qmake; make clean
 
 clean:
@@ -29,13 +26,11 @@ clean:
 	rm -fr zyGrib.app
 	rm -f  src/zyGrib
 	rm -f  src/release/zyGrib.exe
-	cd src/qwt-6.0.1/src; $(QMAKE); make clean
 	cd src;	$(QMAKE); make clean
 
 zyGrib: $(SRC)
 	@ echo
 	rm -f ./zyGrib
-	cd src/qwt-6.0.1/src; $(QMAKE); make -j5
 	cd src; $(QMAKE); make -j5 LFLAGS=" -Wl,--as-needed"
 	@ echo "-----------------------------------"
 	@ echo "#!/bin/bash" >> ./zyGrib
Index: zygrib-7.0.0/src/zyGrib.pro
===================================================================
--- zygrib-7.0.0.orig/src/zyGrib.pro
+++ zygrib-7.0.0/src/zyGrib.pro
@@ -1,5 +1,5 @@
 CONFIG += qt release
-# CONFIG += qt debug console
+# CONFIG += qt debug console
 
 TEMPLATE = app
 TARGET   = zyGrib
@@ -11,7 +11,7 @@ INCLUDEPATH += . util map GUI curvedrawe
 # platform specific
 # ----------------------------------------------------
 win32 {
-	INCLUDEPATH += C:/libs/include/
+	INCLUDEPATH += C:/libs/include/ .. /usr/include/qwt
 	LIBS += -LC:/libs/lib/ -lbz2 -lz -lproj -lnova -lqwt
 	DESTDIR = release
 	RC_FILE += resource.rc
@@ -44,7 +44,7 @@ OBJECTS_DIR = objs
 MOC_DIR = objs
 UI_DIR  = GUI
 
-QT += network xml
+QT += network xml widgets printsupport
 
 CODECFORTR = UTF-8
 CODECFORSRC = UTF-8
@@ -109,10 +109,10 @@ HEADERS += \
            FileLoaderGRIB.h \
            FileLoaderIAC.h \
            FileLoaderMBLUE.h \
-           Font.h \
-           GshhsRangsReader.h \
-           GshhsReader.h \
-           GisReader.h \
+           util/Font.h \
+           map/GshhsRangsReader.h \
+           map/GshhsReader.h \
+           map/GisReader.h \
            GribAnimator.h \
            GribPlot.h \
            GribReader.h \
@@ -138,19 +138,19 @@ HEADERS += \
            MainWindow.h \
            MapDrawer.h \
            MenuBar.h \
-           Orthodromie.h \
-           POI.h \
-           POI_Editor.h \
-           PositionEditor.h \
-           Projection.h \
-           Settings.h \
+           util/Orthodromie.h \
+           map/POI.h \
+           map/POI_Editor.h \
+           map/PositionEditor.h \
+           map/Projection.h \
+           util/Settings.h \
            SkewT.h \
-           SylkFile.h \
+           util/SylkFile.h \
            Terrain.h \
            Therm.h \
-           Util.h \
+           util/Util.h \
            Version.h \
-           zuFile.h
+           util/zuFile.h
 
 SOURCES += \
 			 GUI/PositionEditorWidget.cpp \
@@ -188,15 +188,15 @@ SOURCES += \
            FileLoaderGRIB.cpp \
            FileLoaderIAC.cpp \
            FileLoaderMBLUE.cpp \
-           Font.cpp \
+           util/Font.cpp \
 		   GriddedPlotter.cpp \
 		   GriddedReader.cpp \
 		   GriddedRecord.cpp \
-           GshhsRangsReader.cpp \
-           GshhsReader.cpp \
+           map/GshhsRangsReader.cpp \
+           map/GshhsReader.cpp \
            GribAnimator.cpp \
            GribPlot.cpp \
-           GisReader.cpp \
+           map/GisReader.cpp \
            GribReader.cpp \
            GribRecord.cpp \
            IacPlot.cpp \
@@ -215,18 +215,18 @@ SOURCES += \
            MeteoTable.cpp \
            MeteoTableWidget.cpp \
            MeteotableOptionsDialog.cpp \
-           Orthodromie.cpp \
-           POI.cpp \
-           POI_Editor.cpp \
-           PositionEditor.cpp \
-           Projection.cpp \
-           Projection_libproj.cpp \
-           Settings.cpp \
+           util/Orthodromie.cpp \
+           map/POI.cpp \
+           map/POI_Editor.cpp \
+           map/PositionEditor.cpp \
+           map/Projection.cpp \
+           map/Projection_libproj.cpp \
+           util/Settings.cpp \
            SkewT.cpp \
            SkewTWindow.cpp \
            Terrain.cpp \
            Therm.cpp \
-           Util.cpp \
-           zuFile.cpp
+           util/Util.cpp \
+           util/zuFile.cpp
 
 
Index: zygrib-7.0.0/src/DialogSelectMetar.cpp
===================================================================
--- zygrib-7.0.0.orig/src/DialogSelectMetar.cpp
+++ zygrib-7.0.0/src/DialogSelectMetar.cpp
@@ -179,7 +179,7 @@ void DialogSelectMetar::make_metar_tree
 		item->setData (0, Qt::UserRole+1, a.icao);
 	}
 	
-	treeWidget->header()->setResizeMode (QHeaderView::ResizeToContents);
+	treeWidget->header()->setSectionResizeMode (QHeaderView::ResizeToContents);
 }
 //=============================================================================
 // GUI
Index: zygrib-7.0.0/src/GUI/ColorEditorWidget.cpp
===================================================================
--- zygrib-7.0.0.orig/src/GUI/ColorEditorWidget.cpp
+++ zygrib-7.0.0/src/GUI/ColorEditorWidget.cpp
@@ -1,4 +1,5 @@
 #include <QtGui>
+#include <QColorDialog>
 
 #include "ColorEditorWidget.h"
 
Index: zygrib-7.0.0/src/GUI/LineEditorWidget.cpp
===================================================================
--- zygrib-7.0.0.orig/src/GUI/LineEditorWidget.cpp
+++ zygrib-7.0.0/src/GUI/LineEditorWidget.cpp
@@ -1,4 +1,5 @@
 #include <QtGui>
+#include <QColorDialog>
 
 #include "LineEditorWidget.h"
 
Index: zygrib-7.0.0/src/GUI/TextStyleEditorWidget.cpp
===================================================================
--- zygrib-7.0.0.orig/src/GUI/TextStyleEditorWidget.cpp
+++ zygrib-7.0.0/src/GUI/TextStyleEditorWidget.cpp
@@ -1,4 +1,6 @@
 #include <QtGui>
+#include <QFontDialog>
+#include <QColorDialog>
 
 #include "TextStyleEditorWidget.h"
 
Index: zygrib-7.0.0/src/curvedrawer/CurveDrawer.h
===================================================================
--- zygrib-7.0.0.orig/src/curvedrawer/CurveDrawer.h
+++ zygrib-7.0.0/src/curvedrawer/CurveDrawer.h
@@ -12,10 +12,10 @@
 #include <ctime>
 #include <QObject>
 
-#include <QtGui/QAction>
-#include <QtGui/QToolBar>
-#include <QtGui/QLabel>
-#include <QtGui/QScrollArea>
+#include <QAction>
+#include <QToolBar>
+#include <QLabel>
+#include <QScrollArea>
 
 #include "GriddedPlotter.h"
 #include "DataDefines.h"
@@ -24,14 +24,17 @@
 #include "CustomQwtClasses.h"
 //#include "Polar.h"
 
-#include <qwt_plot_curve.h>
-#include <qwt_scale_engine.h>
-#include <qwt_plot_picker.h>
-#include <qwt_picker_machine.h>
-#include <qwt_legend_item.h>
-#include <qwt_plot_marker.h>
+#include <qwt/qwt_plot_curve.h>
+#include <qwt/qwt_scale_engine.h>
+#include <qwt/qwt_plot_picker.h>
+#include <qwt/qwt_picker_machine.h>
+#include <qwt/qwt_legend.h>
+#include <qwt/qwt_legend_label.h>
+#include <qwt/qwt_legend_data.h>
+#include <qwt/qwt_plot_canvas.h>
+#include <qwt/qwt_plot_marker.h>
 
-#include "qwt_plot.h"
+#include <qwt/qwt_plot.h>
 
 
 //=====================================================================================
Index: zygrib-7.0.0/src/curvedrawer/CustomQwtClasses.h
===================================================================
--- zygrib-7.0.0.orig/src/curvedrawer/CustomQwtClasses.h
+++ zygrib-7.0.0/src/curvedrawer/CustomQwtClasses.h
@@ -10,10 +10,11 @@
 #ifndef CUSTOMQWTCLASSES_H
 #define CUSTOMQWTCLASSES_H
 
-#include <qwt_scale_draw.h>
-#include <qwt_plot_picker.h>
-#include <qwt_picker_machine.h>
-#include <qwt_plot_curve.h>
+#include <qwt/qwt_scale_draw.h>
+#include <qwt/qwt_plot_picker.h>
+#include <qwt/qwt_picker_machine.h>
+#include <qwt/qwt_plot_canvas.h>
+#include <qwt/qwt_plot_curve.h>
 #include <QPainter>
 #include <QVector>
 #include <QDateTime>
@@ -43,7 +44,7 @@ class CustomQwtPicker:
 	public QwtPlotPicker 
 {
 public:
-	CustomQwtPicker( int, int, RubberBand, QwtPickerMachine*, DisplayMode, QwtPlotCanvas*, const QDateTime&, QwtPlotCurve* );
+	CustomQwtPicker( int, int, RubberBand, QwtPickerMachine*, DisplayMode, QWidget*, const QDateTime&, QwtPlotCurve* );
 
 protected:
 	virtual QwtText trackerTextF( const QPointF& ) const;
Index: zygrib-7.0.0/src/main.cpp
===================================================================
--- zygrib-7.0.0.orig/src/main.cpp
+++ zygrib-7.0.0/src/main.cpp
@@ -36,9 +36,10 @@ int main (int argc, char *argv[])
 {
     QApplication app(argc, argv);
 	qsrand(QTime::currentTime().msec());
-	
-    QTextCodec::setCodecForTr(QTextCodec::codecForName("utf8"));
-    QTextCodec::setCodecForCStrings(QTextCodec::codecForName("utf8"));
+
+    // Removed in QT5. utf8 is default anyway	
+    // QTextCodec::setCodecForTr(QTextCodec::codecForName("utf8"));
+    // QTextCodec::setCodecForCStrings(QTextCodec::codecForName("utf8"));
 
 #ifdef Q_OS_MACX
     if ( QSysInfo::MacintoshVersion > QSysInfo::MV_10_8 )
Index: zygrib-7.0.0/src/curvedrawer/CurveDrawer.cpp
===================================================================
--- zygrib-7.0.0.orig/src/curvedrawer/CurveDrawer.cpp
+++ zygrib-7.0.0/src/curvedrawer/CurveDrawer.cpp
@@ -399,7 +399,7 @@ void CurveDrawer::initDataPlot( const in
 									QwtPlotPicker::CrossRubberBand,
 									new QwtPickerClickRectMachine,
 									QwtPicker::ActiveOnly,									
-									qwtDataPlot->canvas(), dt,
+									qobject_cast<QwtPlotCanvas*>(qwtDataPlot->canvas()) , dt,
 									qwtCurve[0]);
 	// connect signals
 //	connect(qwtPicker, SIGNAL(selected(const QwtDoublePoint &)), qwtPicker, SLOT(selected(const QwtDoublePoint &)));
@@ -407,7 +407,7 @@ void CurveDrawer::initDataPlot( const in
 	//------------------------------------------------------------------
 	// set up legend, activate curves
 	//
-	qwtLegend->setItemMode( QwtLegend::CheckableItem );
+	qwtLegend->setDefaultItemMode( QwtLegendData::Checkable );
 	qwtLegend->setLineWidth(10);
 	qwtDataPlot->insertLegend( qwtLegend, QwtPlot::BottomLegend );
 	activateLegend( qwtCurve[0], true );
@@ -612,9 +612,11 @@ void CurveDrawer::activateLegend( QwtPlo
 		pItem->setVisible( on );
 	}
 	// check legend button
-	QWidget *qWid = qwtDataPlot->legend()->find( pItem );
-	if( qWid && qWid->inherits( "QwtLegendItem" ) )
-		((QwtLegendItem *)qWid)->setChecked( on );
+	// QwtLegend* leg = qobject_cast<QwtLegend*>(qwtDataPlot->legend());
+	// leg->checkBox->setCheckState( on );
+	// QWidget *qWid = qwtDataPlot->legend()->find( pItem );
+	// QWidget *qWid = leg->find(pItem);;
+	// if( qWid && qWid->inherits( "QwtLegend" ) )
 	
 }
 //-------------------------------------------------------------------------------
Index: zygrib-7.0.0/src/curvedrawer/CustomQwtClasses.cpp
===================================================================
--- zygrib-7.0.0.orig/src/curvedrawer/CustomQwtClasses.cpp
+++ zygrib-7.0.0/src/curvedrawer/CustomQwtClasses.cpp
@@ -7,7 +7,7 @@
  */
 #include "CustomQwtClasses.h"
 #include "Util.h"
-
+#include <qwt/qwt_point_data.h>
 /*
  *=========================================================================================
  *   Custom Scale Draw Class
@@ -52,7 +52,7 @@ QwtText CustomQwtScale::label( double va
 // Constructor for custom picker class with current date and picker machine
 //-------------------------------------------------------------------------------
 CustomQwtPicker::CustomQwtPicker( int xAxis, int yAxis, RubberBand rubberBand, QwtPickerMachine *pickerMachine, DisplayMode trackerMode,
-								 QwtPlotCanvas *canvas, const QDateTime &dtCurrent, QwtPlotCurve *qwtCurve )
+								 QWidget *canvas, const QDateTime &dtCurrent, QwtPlotCurve *qwtCurve )
 	: QwtPlotPicker( xAxis, yAxis, rubberBand, trackerMode, canvas )
 {
 	dtCurPos = dtCurrent;
Index: zygrib-7.0.0/src/map/POI.cpp
===================================================================
--- zygrib-7.0.0.orig/src/map/POI.cpp
+++ zygrib-7.0.0/src/map/POI.cpp
@@ -137,10 +137,10 @@ void POI::readSettings (uint code, bool
 	isMovable = Settings::getSettingPOI (code, "move", false, fnat).toBool();
 	showLabel = Settings::getSettingPOI (code, "showLabel", true, fnat).toBool();
 	setDisplayParams
-		( Settings::getSettingPOI (code, "markColor", Qt::red, fnat).value<QColor>(),
+		( Settings::getSettingPOI (code, "markColor", QColor(Qt::red), fnat).value<QColor>(),
 	      Settings::getSettingPOI (code, "labelFont", Font::getFont(FONT_POILabel), fnat).value<QFont>(),
-		  Settings::getSettingPOI (code, "labelTextColor", Qt::black, fnat).value<QColor>(),
-		  Settings::getSettingPOI (code, "labelBgColor", Qt::white, fnat).value<QColor>()
+		  Settings::getSettingPOI (code, "labelTextColor", QColor(Qt::black), fnat).value<QColor>(),
+		  Settings::getSettingPOI (code, "labelBgColor", QColor(Qt::white), fnat).value<QColor>()
 		);		
 }
 //-------------------------------------------------------------------------------
